src/toil/job.py

```diff
@@ -1402,41 +2351,61 @@ def _executor(self, jobGraph, stats, fileStore):
                 )
             )
 
-    def _runner(self, jobGraph, jobStore, fileStore, defer):
+    def _runner(self, jobStore=None, fileStore=None, defer=None, **kwargs):
         """
         This method actually runs the job, and serialises the next jobs.
+        
+        It marks the job as completed (by clearing its command) and creates the
+        successor relationships to new successors, but it doesn't actually
+        commit those updates to the current job into the JobStore.
+
+        We take all arguments as keyword arguments, and accept and ignore
+        additional keyword arguments, for compatibility with workflows (*cough*
+        Cactus *cough*) which are reaching in and overriding _runner (which
+        they aren't supposed to do). If everything is passed as name=value it
+        won't break as soon as we add or remove a parameter.
 
-        :param class jobGraph: Instance of a jobGraph object
         :param class jobStore: Instance of the job store
-        :param toil.fileStores.abstractFileStore.AbstractFileStore fileStore: Instance of a cached
-               or uncached filestore
+        :param toil.fileStores.abstractFileStore.AbstractFileStore fileStore: Instance
+               of a cached or uncached filestore
         :param defer: Function yielded by open() context
                manager of :class:`toil.DeferredFunctionManager`, which is called to
                register deferred functions.
+        :param kwargs: Catch-all to accept superfluous arguments passed by old
```

Comment:
Can we create an issue in Cactus for this (and a subsequent toil issue to remove this after this PR is merged)?  I really dislike having this and it may cause unexpected errors on cactus's end if they're expecting certain behavior from their placebo kwargs.

Reply:
I have the Cactus PR and issue already I think, but I opened https://github.com/DataBiosphere/toil/issues/3302 here for the cleanup after Cactus changes and people upgrade.

Cactus doesn't do anything with the args right now; they just insist on having them to hand back to us.